components/script/dom/htmlmediaelement.rs

```diff
@@ -1220,6 +1234,118 @@ impl HTMLMediaElement {
             },
             PlayerEvent::MetadataUpdated(ref metadata) => {
                 // https://html.spec.whatwg.org/multipage/#media-data-processing-steps-list
+                // => If the media resource is found to have an audio track
+                if !metadata.audio_tracks.is_empty() {
+                    for (i, _track) in metadata.audio_tracks.iter().enumerate() {
+                        // Step 1.
+                        let kind = match i {
+                            0 => "main",
+                            _ => "",
+                        };
+                        let window = window_from_node(self);
+                        let audio_track = AudioTrack::new(
+                            &window,
+                            DOMString::new(),
+                            DOMString::from(kind),
+                            DOMString::new(),
+                            DOMString::new(),
+                        );
+
+                        // Step 2.
+                        self.AudioTracks().add(&audio_track);
+
+                        // Step 3.
+                        audio_track.set_enabled(false);
+
+                        // Step 4
+                        // https://www.w3.org/TR/media-frags/#media-fragment-syntax
+                        // https://github.com/servo/servo/issues/22366
+
+                        // Step 5. & 6,
+                        if audio_track.enabled() || self.AudioTracks().enabled_index().is_none() {
+                            let window = window_from_node(self);
+                            let task_source = window.task_manager().media_element_task_source();
+                            let this = Trusted::new(self);
+
+                            task_source
```

Comment:
Is this the correct way to implement the task source event? I am not familiar with the naming convention for the tasks or if this is specific in some way.

Reply:
Some other areas will also queue this task and I am wondering whether these should take place in AudioTrackList and VideoTrackList?